feature/search/src/main/java/com/napzak/market/search/SearchScreen.kt

```diff
@@ -0,0 +1,278 @@
+import com.napzak.market.designsystem.component.textfield.SearchTextField
+import com.napzak.market.designsystem.theme.NapzakMarketTheme
+import com.napzak.market.feature.search.R.drawable.ic_left_chevron
+import com.napzak.market.feature.search.R.string.search_hint
+import com.napzak.market.feature.search.R.string.search_suggested_search_text
+import com.napzak.market.feature.search.R.string.search_suggested_genre
+import com.napzak.market.search.component.GenreNavigationButton
+import com.napzak.market.search.component.SuggestedGenreCard
+import com.napzak.market.search.component.SuggestedKeywordChip
+import com.napzak.market.util.android.noRippleClickable
+
+const val EMPTY_TEXT = ""
+
+@Composable
+internal fun SearchRoute(
+    onBackButtonClick: () -> Unit,
+    onSearchResultNavigate: (String) -> Unit,
+    onGenreDetailNavigate: (Long) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    var searchText by remember { mutableStateOf("") } // TODO: viewModel로 옮길 예정
+
+    SearchScreen(
+        searchText = searchText,
+        suggestedSearchText = emptyList(),
+        suggestedGenre = emptyList(),
+        searchResultGenres = emptyList(),
+        onBackButtonClick = onBackButtonClick,
+        onTextChange = { searchText = it },
+        onSearchClick = { onSearchResultNavigate(searchText) },
+        onSuggestedTextClick = onSearchResultNavigate,
+        onSuggestedGenreClick = onGenreDetailNavigate,
+        modifier = modifier,
+    )
+}
+
+@Composable
+private fun SearchScreen(
+    searchText: String,
+    suggestedSearchText: List<String>,
+    suggestedGenre: List<Genre>,
+    searchResultGenres: List<Genre>,
+    onBackButtonClick: () -> Unit,
+    onTextChange: (String) -> Unit,
+    onSearchClick: () -> Unit,
+    onSuggestedTextClick: (String) -> Unit,
+    onSuggestedGenreClick: (Long) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(color = NapzakMarketTheme.colors.gray10),
+    ) {
+        Row(
+            modifier = Modifier
+                .shadow(
+                    elevation = 4.dp,
+                    spotColor = NapzakMarketTheme.colors.transBlack,
+                    ambientColor = NapzakMarketTheme.colors.transBlack,
+                )
+                .background(color = NapzakMarketTheme.colors.white)
+                .padding(start = 20.dp, top = 66.dp, end = 20.dp, bottom = 20.dp),
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Icon(
+                imageVector = ImageVector.vectorResource(ic_left_chevron),
+                contentDescription = null,
+                tint = Color.Unspecified,
+                modifier = Modifier.noRippleClickable(onBackButtonClick),
+            )
+
+            Spacer(Modifier.width(12.dp))
+
+            SearchTextField(
+                text = searchText,
+                onTextChange = { onTextChange(it) },
```

Comment:
```suggestion
                onTextChange = onTextChange,
```
p2: 이렇게 바꿀 수 있지 않나요??